File: Low Level Design (LLD)/Problems/05. FoodDeliverySystem/Java/service/UserService.java
package service;

import models.User;

import java.util.HashMap;
import java.util.Map;

public class UserService {
    private Map<String, User> users = new HashMap<>();

    public void registerUser(User user) {
        users.put(user.getId(), user);
    }

    public User getUserById(String id) {
        return users.get(id);
    }
}

File: Low Level Design (LLD)/Problems/05. FoodDeliverySystem/Cpp/service/src/UserService.cpp
#include "../include/UserService.h"

void UserService::registerUser(const std::shared_ptr<User>& user) {
    users[user->getId()] = user;
}

const std::shared_ptr<const User> UserService::getUserById(const std::string& id) const {
    auto it = users.find(id);
    if (it != users.end()) {
        return it->second;
    }
    return nullptr;
}